### DISCUSSION TITLE
Node Unassignment Reason

### DISCUSSION THREAD

**Discussion by PranoyKumar3** [google/or-tools#2362] [Routing (and legacy CP) questions]
**What language and solver does this apply to?**
Python /  Routing

**Describe the problem you are trying to solve.**
Google OR-Tools has a feature to make visiting node optional by using AddDisjunction, which gives the solver the freedom to drop a node if it's unable to satisfy a constraint. But is it possible to know due to which constraint violation was the node dropped?
(In case of CVRPTW, the reason for dropping a node could either be that vehicle max capacity has reached or the node could not be reached within the particular time-window.)
**Describe the solution you'd like**
It would be good if there was a feature which could address the unassigned nodes along with the unassignment reason.
**Describe alternatives you've considered**

**Additional context**

**Comment by lperron** [comment]
The node could be dropped because the solver did not manage to insert it, even it if was possible.
